develop/CICD/Platform_Independent_CICD.md

```diff
@@ -0,0 +1,100 @@
+
+- [Skyline.DataMiner.CICD.Tools.NuGetPackageConfigDetector](https://www.nuget.org/packages/Skyline.DataMiner.CICD.Tools.NuGetPackageConfigDetector#readme-body-tab)
+
+Most tools and code that need to read or change NuGet libraries in a solution only work either on Legacy style package.config or SDK Style NuGet usage. This dotnet tool will check if every project in a Visual Studio solution is SDK style or Legacy style. Calling it while providing the path to the Workspace will return a "#"-separated list with all project names still using the legacy style packages.config.
+
+- [Skyline.DataMiner.CICD.Tools.NuGetToggleOnBuild](https://www.nuget.org/packages/Skyline.DataMiner.CICD.Tools.NuGetToggleOnBuild#readme-body-tab)
+
+When creating libraries that end up as NuGets, you'll often let MSBuild automatically create the NuGets. However during a pipeline run you often need to run MSBuild more than once. This tool allows you to disable or enable creation of the NuGets during MSBuild. This helps in avoiding a pipeline to create the NuGets more than once.
+
+- [Skyline.DataMiner.CICD.Tools.NuGetPreBuildApplyBranchOrTag](https://www.nuget.org/packages/Skyline.DataMiner.CICD.Tools.NuGetPreBuildApplyBranchOrTag/1.0.1-1.0.0.X.20#readme-body-tab)
+
+When creating libraries that end up as NuGets, this tool will change the version of the to be created NuGets and Assemblies in your solution to the specific version (this is often the Tag you provide in GIT). It also allows you to provide the name of a branch with a buildnumber which will then create a pre-release version from that data.
+
+- [Skyline.DataMiner.CICD.Tools.NuGetChangeVersion](https://www.nuget.org/packages/Skyline.DataMiner.CICD.Tools.NuGetChangeVersion#readme-body-tab)
+
+This allows you to change the version of a NuGet used in your SDK solution to a specific version (or the highest one).
+
+For the complete list, use NuGet and search for [Skyline CICD.Tools](https://www.nuget.org/packages?q=Skyline+CICD.Tools&prerel=true&sortby=relevance)
+
+## NuGet Libraries
+
+We also provide code libraries that can be used for development of custom programs. These contain functionality that we use when creating programs like the .NET Tools or DIS.
+By providing these, it allows flexibility to other developers if they need to create their own extensions, programs or tools specifically for your pipelines.
+
+The following libraries are the most useful:
+
+- [Skyline.DataMiner.CICD.FileSystem](https://www.nuget.org/packages/Skyline.DataMiner.CICD.FileSystem#readme-body-tab)
+
+Recommended to use this instead of System.IO for all tool/stage development in c#. This supports *long paths* in windows and also allows you to manipulate windows style paths while on a linux system and the other way around. It also handles some authentication issues you may encounter with System.IO when running in GitHub.
```

Comment:
```suggestion
Recommended to use this instead of System.IO for all tool/stage development in C#. This supports *long paths* in windows and also allows you to manipulate windows style paths while on a Linux system and the other way around. It also handles some authentication issues you may encounter with System.IO when running on GitHub.
```